### PR TITLE
Allow setting controllers and a couple other customizations

### PR DESCRIPTION
**Pull request by anqit**

### CODE REVIEW

File: lib/multiple_search_selection.dart [hunk 1]
```diff
@@ -739,8 +758,8 @@ class _MultipleSearchSelectionState<T>
         if (widget.title != null) ...[
           widget.title!,
         ],
-        if (pickedItems.isNotEmpty)
-          widget.pickedItemsContainerBuilder?.call([
+        if (widget.pickedItemsContainerBuilder != null)
```
Comment (anqit) [inline lib/multiple_search_selection.dart:761]: i wanted to use my widget even when the picked items list was empty

File: lib/multiple_search_selection.dart [hunk 2]
```diff
@@ -565,11 +589,6 @@ class _MultipleSearchSelectionState<T>
   bool expanded = false;
 
   List<T> pickedItems = [];
-  final ScrollController _pickedItemsController = ScrollController();
-  final ScrollController _showedItemsScrollController = ScrollController();
-  final TextEditingController _textEditingController = TextEditingController();
-
-  final FocusNode _textFieldFocus = FocusNode();
```
[the remark this answers is not in the document]
Comment (anqit) [reply lib/multiple_search_selection.dart:572]: in fact, i'm curious enough to try to get some expert answers https://stackoverflow.com/questions/75356989/initializing-non-mutable-fields-in-stateful-widgets-in-flutter

[the remark this answers is not in the document]
Comment (esentis) [reply lib/multiple_search_selection.dart:572]: Well for non-mutable fields I don't see why we should re-create the variable in the widget's state. I always prefer accessing it by `widget.`.

File: lib/multiple_search_selection.dart [hunk 3]
```diff
@@ -46,7 +46,7 @@ class MultipleSearchSelection<T> extends StatefulWidget {
     bool? showShowedItemsScrollbar,
     bool? showSelectAllButton,
     bool? showClearAllButton,
-    InputDecoration? searchFieldInputDecoration,
```
Comment (esentis) [inline lib/multiple_search_selection.dart:49]: Why have you remove this one ?

File: lib/multiple_search_selection.dart (hunk 1 above)
Comment (esentis) [reply lib/multiple_search_selection.dart:761]: Ok this makes sense.

**Review by esentis — CHANGES_REQUESTED** [review]
We can keep the changes with the initializers and removing the state variables. However, please explain me why you removed the `decoration` of search field.

File: lib/multiple_search_selection.dart (hunk 3 above)
Comment (anqit) [reply lib/multiple_search_selection.dart:49]: Sorry, I don't think I meant to submit this deletion, I was experimenting with trying to get the clear text field button working and must have removed this then. I will fix...

**Comment by anqit** [comment]
ok, should be fixed

**Review by esentis — APPROVED** [review]

**Comment by esentis** [comment]
Great everything looks good, will be released with the next version. Next version will also refactor how we render the showedItems to be lazily build instead of `ListView` & `shrinkWrap` since it effects performence when we are dealing with large lists.

**Pull request merged**